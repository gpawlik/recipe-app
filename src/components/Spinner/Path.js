import styled, { keyframes } from 'styled-components';

import { colors } from 'variables';

const offset = 100;

const dash = keyframes`
  0% { stroke-dashoffset: ${offset}; }
  50% {
    stroke-dashoffset: ${offset}/4;
    transform:rotate(135deg);
  }
  100% {
    stroke-dashoffset: ${offset};
    transform:rotate(450deg);
  }
`;

const Path = styled.circle`
  stroke: ${colors.GREEN_SEA};
  stroke-width: 6;
  stroke-linecap: round;
  stroke-dasharray: ${offset};
  stroke-dashoffset: 0;
  transform-origin: center;
  animation: ${dash} 1.4s ease-in-out infinite;
`;

export default Path;
