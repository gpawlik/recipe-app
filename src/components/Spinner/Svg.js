import styled, { keyframes } from 'styled-components';

const rotator = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(270deg); }
`;

const Svg = styled.svg`
  animation: ${rotator} 1.4s linear infinite;
`;

export default Svg;
