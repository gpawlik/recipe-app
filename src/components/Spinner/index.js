import React from 'react';
import PropTypes from 'prop-types';

import Path from './Path';
import Svg from './Svg';

const Spinner = ({ width, height }) => {
  return (
    <Svg width={width} height={height} viewBox="0 0 66 66">
      <Path fill="none" stroke-linecap="round" cx={width / 2} cy={height / 2} r="30"></Path>
    </Svg>
  );
};

Spinner.defaultProps = {
  width: 66,
  height: 66
};

Spinner.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number
};

export default Spinner;
