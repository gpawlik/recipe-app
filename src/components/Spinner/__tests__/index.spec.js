import React from 'react';
import { create } from 'react-test-renderer';

import Spinner from '../';

const initialProps = {};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <Spinner {...finalProps} />
  );
};

describe('Spinner', () => {
  it('should render with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with custom width and height', () => {
    const props = {
      width: 10,
      height: 10
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
