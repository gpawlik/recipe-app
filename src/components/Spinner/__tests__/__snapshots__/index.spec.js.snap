// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render with custom width and height 1`] = `
<svg
  className="sc-bwzfXH bEKcUv"
  height={10}
  viewBox="0 0 66 66"
  width={10}
>
  <circle
    className="sc-bdVaJa eiyToN"
    cx={5}
    cy={5}
    fill="none"
    r="30"
  />
</svg>
`;

exports[`Spinner should render with the default props 1`] = `
<svg
  className="sc-bwzfXH bEKcUv"
  height={66}
  viewBox="0 0 66 66"
  width={66}
>
  <circle
    className="sc-bdVaJa eiyToN"
    cx={33}
    cy={33}
    fill="none"
    r="30"
  />
</svg>
`;
